### PR TITLE
Add versup to project for versioning

### LINKED ISSUE #23: Replace versioning tool with an opensource alternative
The versioning tool used (Usurp) is not available elsewhere. Suggest using an opensource alternative, like the one you have also been working on

https://github.com/Svenito/versup

### PR DESCRIPTION
**Pull request by Svenito**
This adds [Versup](https://github.com/Svenito/versup) to the project to replace the closed source usurp with an opensource alternative.

This commit adds a dependency to the `setup.py` file and an initial config file replacing the current usurp config. The config will not create a changelog and will replace the version numbers in `release_nodes.rst`, and `source/wiz/_version.py` files as usurp did.

The existing tags work already with versup, so the next version can be set by running

    versup patch

or whatever increment it needs. To ensure you are going to get what you expect, you can run

    versup patch -n

as a dry run. It will inform you of all changes that are going to be made.

Closes #23

### CODE REVIEW

**Comment by buddly27** [comment]
So I'm running into an issue when running the dry-run function, but usurp has the same problem:
```
(dev-py27) mb647:wiz > versup patch -n
Traceback (most recent call last):
  File "/Users/jeremyr/.virtualenvs/dev-py27/bin/versup", line 8, in <module>
    sys.exit(main())
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/versup/__main__.py", line 5, in main
    versup.command_line.main()
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/versup/command_line.py", line 225, in main
    cli()
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/versup/command_line.py", line 88, in do_versup
    template.token_data["author_name"] = gitops.get_username()
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/versup/gitops.py", line 12, in get_username
    return repo.config_reader().get_value("user", "name")
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/git/config.py", line 590, in get_value
    valuestr = self.get(section, option)
  File "/Users/jeremyr/.virtualenvs/dev-py27/lib/python2.7/site-packages/git/config.py", line 79, in assure_data_present
    return func(self, *args, **kwargs)
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ConfigParser.py", line 330, in get
    raise NoSectionError(section)
ConfigParser.NoSectionError: No section: 'user'
```

The issue seems to be that I'm using the "includeIf" condition in my `.gitconfig` to set the user depending on the current directory. GitPython doesn't know how to read that...
```
[includeIf "gitdir:~/dev/mill/"]
        path = ~/.gitconfig.mill
[includeIf "gitdir:~/dev/github/"]
        path = ~/.gitconfig.perso
```

So I guess it's not a blocker, but I'd be interested if you have any idea how to fix that :)

File: setup.py [hunk 1]
```diff
@@ -32,7 +32,8 @@
     "sawmill >= 0.2.1, < 1",
     "six >= 1.15.0, < 2",
     "toml >= 0.10.1, < 1",
-    "ujson >= 2.0.3, < 4"
+    "ujson >= 2.0.3, < 4",
+    "versup >= 1.0.1",
```
Comment (buddly27) [inline setup.py:36]: The dependency is not really necessary for the package installation, but then it doesn't really belong in "doc" and "test" either.

What do you think about adding it in an extra list that would be added to the "dev" extra requirement?

```python
DEV_REQUIRES = [
    "versup >= 1.0.1",
]

...

setup(
    ...
    extras_require={
        "doc": DOC_REQUIRES,
        "test": TEST_REQUIRES,
        "dev": DOC_REQUIRES + TEST_REQUIRES + DEV_REQUIRES
    },
   ...
)
```

File: .versup.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        ],
+        "doc/release/release_notes.rst": [
+            [
+                ".. release:: Upcoming",
+                ".. release:: [version]\n    :date: [version_date]"
+            ]
+        ]
+    },
+    "changelog": {
+        "create": false
```
Comment (buddly27) [inline .versup.json:14]: It doesn't seem to work, the changelog is still being generated:

```
(dev-py27) mb647:wiz > versup patch -n
In file source/wiz/_version.py replace __version__ = "3.1.2" with __version__ = "3.1.3"
✓ Updated doc/release/release_notes.rst, source/wiz/_version.py

Writing changelog entries:

- Fix incorrect type in versup file
- Add versup to project for versioning


✓ Changelog updated

Create commit with commit msg: Update version to 3.1.3
✓ Commit created

Create tag 3.1.3 with msg v3.1.3
✓ Tag v3.1.3 created
```

File: setup.py (hunk 1 above)
Comment (Svenito) [reply setup.py:36]: Yes, ideally it should be a dev dependency, but I didn't want to add a new section. I'll make that change

File: .versup.json (hunk 2 above)
Comment (Svenito) [reply .versup.json]: Fixed in recent commit 2686760

**Comment by Svenito** [comment]
> So I'm running into an issue when running the dry-run function, but usurp has the same problem:
> I guess it's not a blocker, but I'd be interested if you have any idea how to fix that :)

Yes, it looks like the configparser from GitPython doesn't parse the config from the conditional include which is really annoying.
I've raised a ticket  gitpython-developers/GitPython#1053 for this. Otherwise I have no workaround other than creating a default user block in the global or at least local repo config.

File: setup.py (hunk 1 above)
Comment (buddly27) [reply setup.py:36]: Looks good, could you remove "versup" from the installation requirements?

Also could you add an extra blank line between `DEV_REQUIRES` and `BuildExtended`? 
https://www.python.org/dev/peps/pep-0008/#blank-lines

As discussed previously, this is where [black](https://github.com/psf/black) would definitely be useful! (#36)

File: .versup.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "files": {
+        "source/wiz/_version.py": [
+            ["__version__ = \"([\\d\\.]+)\"", "__version__ = \"[version]\""]
```
Comment (buddly27) [inline .versup.json:4]: Could you reformat this block as follows to improve readability?

```
{
    "files": {
        "source/wiz/_version.py": [
            [
                "__version__ = \"([\\d\\.]+)\"",
                "__version__ = \"[version]\""
            ]
        ]
       ...
}
```

Comment (buddly27) [reply .versup.json:4]: Oh and also, please remove `usurp.json` as it won't be required anymore :)

**Comment by buddly27** [comment]
I updated the target to `master` so that I can release it tomorrow once you updated the branch

**Review by buddly27 — APPROVED** [review]
Looks good, you forgot to add a release note but I'm just gonna merge and add it in master. Thanks!

**Pull request merged**